From 0992e2fdd2e82ce280269302aaa4c5ad0ca34f79 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Tue, 31 Oct 2023 12:48:59 +0530
Subject: [PATCH] Do not report incompatibe relocation table as an error

Specifically, .llvm.call-graph-profile is always REL even on x86-64.
We can safely ignore the relocation table because it contains only
R_NONE relocations.

Fixes #653

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 elf/input-files.cc | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/elf/input-files.cc b/elf/input-files.cc
index 870dfcc..3b404e8 100644
--- a/elf/input-files.cc
+++ b/elf/input-files.cc
@@ -249,15 +249,7 @@ void ObjectFile<E>::initialize_sections(Context<E> &ctx) {
       break;
     }
     case SHT_REL:
-      if constexpr (E::is_rela)
-        Fatal(ctx) << *this << ": REL-type relocation table is not supported"
-                   << " for this target";
-      break;
     case SHT_RELA:
-      if constexpr (!E::is_rela)
-        Fatal(ctx) << *this <<": RELA-type relocation table is not supported"
-                   << " for this target";
-      break;
     case SHT_SYMTAB:
     case SHT_SYMTAB_SHNDX:
     case SHT_STRTAB:
-- 
2.42.0

